app/waf/index.js

```diff
@@ -14,22 +14,36 @@
  * limitations under the License.
  */
 
+const fs = require('fs');
 const application = require('./dist');
 
 module.exports = application;
 
 if (require.main === module) {
   // Run the application
-  const config = {
+  let config = {
     rest: {
       port: +process.env.WAF_APP_PORT || 3000,
       host: process.env.WAF_APP_HOST || '0.0.0.0',
+      protocol: 'http',
       openApiSpec: {
         // useful when used with OASGraph to locate your application
         setServersFromRequest: true,
       },
     },
   };
+
+  if (process.env.ENABLE_HTTPS) {
+    config.rest['protocol'] = 'https';
+    config.rest['port'] = +process.env.WAF_APP_HTTPS_PORT || 443;
+    config.rest['key'] = fs.readFileSync(
+      process.env.CERT_KEY || './certs/domain.key',
```

Comment:
We are shipping this app in container image. A container image may not ship cert files. Usually user need to provide their own cert files. So you may need a volume path for them to inject cert files.

Reply:
Yes, I agree.